packages/ts-morph/src/compiler/types/Type.ts

```diff
@@ -233,9 +233,19 @@ export class Type<TType extends ts.Type = ts.Type> {
     }
 
     /**
-     * Gets the target type of a type reference if it exists.
+     * Gets the reference type if this type is an instantiated type.
      *
-     * For example, given `Promise<string>` this will return `Promise<T>`.
+     * Specifically:
+     *
+     * - If this type is an instantiated type then returns the reference type.
+     * - If this type is a reference type then returns itself (same reference).
+     * - Otherwise returns `undefined`.
+     *
+     * For example:
+     *
+     * - Given `Promise<string>` this will return `Promise<T>`.
+     * - Given `Promise<T>` this will return the same `Promise<T>`.
+     * - Given `string` this will return `undefined`.
```

Comment:
Before Writing the JSdoc for the throws version I want to align on this one. The other will mostly be a copy except `return undefined` being replaced with `throws` ish.

Reply:
> glossary that explains these terms/concepts in detail.

Maybe we should start one 😅

Reply:
I just looked into what the terms may be:

https://github.com/microsoft/TypeScript/blob/4a9e2be386d558cf66d4bd90591c49498419bcf0/lib/typescript.d.ts#L2534

Maybe let's use "generic type" to refer to the target type and "type reference" to refer to the instantiated type so it aligns with the compiler api. We can use `Promise<T>` as an example of a generic type and `Promise<string>` as an example of a type reference. Sound good?

Reply:
@jasonkuhrt yeah, a lot of stuff in ts-morph is undocumented because I haven't understood a lot of stuff if the compiler api haha.